stl/inc/algorithm

```diff
@@ -3836,12 +3885,87 @@ _CONSTEXPR20 void push_heap(_RanIt _First, _RanIt _Last) {
     _STD push_heap(_First, _Last, less<>());
 }
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // VARIABLE ranges::push_heap
+    template <class _It, class _Pr, class _Pj>
+    constexpr void _Push_heap_by_index(const _It _First, iter_difference_t<_It> _Hole,
+        const iter_difference_t<_It> _Top, iter_value_t<_It>&& _Val, _Pr _Pred, _Pj _Proj) {
+        // percolate _Hole to _Top or where _Val belongs, using _Pred

```

Comment:
Suggestion: the classic comment fragment `, using _Pred` is incomplete for ranges (where `_Proj` is also used), and wasn't very important in the first place. (In C++98, I suppose it was a big deal to be using a `_Pred` instead of `operator<`; now the intention to use comparators seems self-explanatory - especially since now we don't have hardcoded `operator<` implementations in 99% of cases, simply using `less` for centralization.) Occurs below.

Reply:
I'd like to burninate all of these, but it's too big a change for this PR. I'll audit to ensure the `ranges` versions all say `using _Pred and _Proj` as intended, and submit a separate PR with cleanup.

Reply:
Created https://github.com/CaseyCarter/STL/tree/using_pred which I'll submit after this PR merges. (Leaving this unresolved as a reminder.)